components/brave_sync/ui/constants/sync_types.ts

```diff
@@ -17,5 +17,14 @@ export const enum types {
   SYNC_SAVED_SITE_SETTINGS = '@@sync/SYNC_SAVED_SITE_SETTINGS',
   SYNC_BROWSING_HISTORY = '@@sync/SYNC_BROWSING_HISTORY',
   SYNC_SETUP_SYNC_HAVE_CODE = '@@sync/SYNC_SETUP_SYNC_HAVE_CODE',
+  SYNC_SETUP_ERROR = '@@sync/SYNC_SETUP_ERROR',
+  SYNC_RESET_SETUP_ERROR = '@@sync/SYNC_RESET_SETUP_ERROR',
   SYNC_ON_LOG_MESSAGE = '@@sync/SYNC_ON_LOG_MESSAGE'
 }
+
+export type onSyncSetupErrorType =
+  'ERR_SYNC_NO_INTERNET' |
+  'ERR_SYNC_MISSING_WORDS' |
+  'ERR_SYNC_WRONG_WORDS' |
+  'ERR_SYNC_NO_DEVICE_NAME' |
```

Comment:
We need a `ERR_SYNC_INIT_FAILED` which means when users device is online but there is something wrong during the initialization process. could be either client side or server side.
https://github.com/brave/sync/blob/e32fd5828b63b053c427c48ec5139504a9c4e9f2/client/constants/messages.js#L24
https://github.com/brave/sync/blob/e32fd5828b63b053c427c48ec5139504a9c4e9f2/client/sync.js#L249
https://github.com/brave/sync/blob/e32fd5828b63b053c427c48ec5139504a9c4e9f2/client/sync.js#L255
but users don't need to know the detail just general message